tests/e2e/test-iac-virtualmodels.sh

```
@@ -0,0 +1,152 @@
+set -a; source "$REPO/.env"; set +a
+PORT="${IAC_PORT:-18091}"
+NEG_PORT="${IAC_NEG_PORT:-18092}"
+B="http://localhost:$PORT"
+
+PIDF="$WORK/gw.pid"
+# start_gw boots a gateway in unsafe mode (empty master key) with an in-memory
+# model cache, so EVERY boot validates managed config against a cold catalog. It
+# waits for /health (startup validation passed) and then for the async model load
+# to finish ("model registry initialized") so resolve-time chats have a catalog.
+start_gw(){ # uses $VM_ENV (may be empty) and $WORK/config.yaml if present
+  for sp in $(lsof -nP -t -iTCP:$PORT -sTCP:LISTEN 2>/dev/null); do kill "$sp" 2>/dev/null; done
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)